planner/languages/rust/rust_planner.go

```diff
@@ -17,9 +24,82 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	return false
+	cargoTomlPath := filepath.Join(srcDir, cargoToml)
+	return plansdk.FileExists(cargoTomlPath)
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
-	return &plansdk.Plan{}
+	plan, err := p.getPlan(srcDir)
+	if err != nil {
+		fmt.Printf("[WARNING] error in RustPlanner.GetPlan: %s", err)
+	}
+	return plan
+}
+
+func (p *Planner) getPlan(srcDir string) (*plansdk.Plan, error) {
+
+	// NOTE: I'm not convinced we need these packages to be included in the long term.
+	// The link indicates we need libiconv for macOS, but maybe not for linux. For now,
+	// including them by default, but this could be likely optimized once we understand better.
+	//
+	// libiconv due to error:
+	//     ld: library not found for -liconv. clang-11: error: linker command failed with exit code 1
+	// https://discourse.nixos.org/t/nix-shell-rust-hello-world-ld-linkage-issue/17381/2
+	//
+	// gcc due to error:
+	//     linker `cc` not found
+	// https://github.com/NixOS/nixpkgs/issues/103642
+	packages := []string{"rustup", "libiconv", "gcc"}
+
+	version, err := p.rustVersion(srcDir)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	rustupVersion := "stable"
+	if version != nil {
+		rustupVersion = version.Exact()
+	}
+
+	rustupDefaultCmd := fmt.Sprintf("rustup default %s", rustupVersion)
+
+	return &plansdk.Plan{
+		DevPackages:     packages,
+		RuntimePackages: packages,
+		Shell: plansdk.PlanShell{
+			InitHook: rustupDefaultCmd,
```

Comment:
I don't love that the planner needs to use the shell hook – it feels like we're having to go outside the nix infrastructure to set the right version. Is there really no way to do it by specifying the right package instead of calling `rustup`?

Reply:
I couldn't find a `rustup_<version>` package. I also feel this is odd, but not quite sure the best alternative fix for now...